### PR TITLE
Update cal.py

### PR DESCRIPTION
**Pull request by pronoym99**
Convert ```format``` strings to python3.6+ compliant ```f-strings``` wherever possible.

### CODE REVIEW

**Comment by pronoym99** [comment]
Please add the hacktoberfest label to this PR if possible.

**Review by niccokunzmann — CHANGES_REQUESTED** [review]
I am not willing to merge this PR for one reason: The 600 lines diff does not fit the purpose. Everyone else editing at the same time gests merge conflicts. We indent with 4 spaces. Please recreate the PR so that it changes the lines required and that being visible on the PR tab showing the files. Happy to merge it then.

File: CHANGES.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ Changelog
 
 Minor changes:
 
+- Converted all ``format`` strings to ``f-strings`` in cal.py. Ref: #481 [pronoym99]
+
```
Comment (niccokunzmann) [inline CHANGES.rst]: There would be no new line in between. - but that is just prettyness, no need to change.

**Comment by niccokunzmann** [comment]
Thanks for your PR! This is nice that it gets ported to later versions!

File: CHANGES.rst (hunk 1 above)
Comment (pronoym99) [reply CHANGES.rst]: Oh sorry...I resolved the merge conflict at just the time you edited your comment. 😄

**Comment by niccokunzmann** [comment]
I really liked the other PRs but I can not understand this one because I do not want to check over 1000 lines. I hope to merge it soon.

File: src/icalendar/cal.py [hunk 2]
```diff
@@ -434,12 +433,8 @@ def to_ical(self, sorted=True):
     def __repr__(self):
         """String representation of class with all of it's subcomponents.
         """
-        subs = ', '.join([str(it) for it in self.subcomponents])
```
Comment (pronoym99) [inline src/icalendar/cal.py:437]: Replaced this with ```join(<generator expression>)```

**Comment by pronoym99** [comment]
> I am not willing to merge this PR for one reason: The 600 lines diff does not fit the purpose. Everyone else editing at the same time gests merge conflicts. We indent with 4 spaces. Please recreate the PR so that it changes the lines required and that being visible on the PR tab showing the files. Happy to merge it then.

@niccokunzmann did changes in a way that properly renders a diff in the PR tab. Please have a look.

File: src/icalendar/cal.py [hunk 3]
```diff
@@ -612,12 +607,8 @@ def to_tz(self):
...
                 tzname = self._make_unique_tzname(tzname, tznames)
             except KeyError:
-                tzname = '{}_{}_{}_{}'.format(
-                    zone,
-                    component['DTSTART'].to_ical().decode('utf-8'),
-                    component['TZOFFSETFROM'].to_ical(),  # for whatever reason this is str/unicode
-                    component['TZOFFSETTO'].to_ical(),  # for whatever reason this is str/unicode
-                )
+                # for whatever reason this is str/unicode
+                tzname = f"{zone}_{component['DTSTART'].to_ical().decode('utf-8')}_{component['TZOFFSETFROM'].to_ical()}_{component['TZOFFSETTO'].to_ical()}"
```
Comment (angatha) [inline src/icalendar/cal.py]: This line got quite long.

Comment (pronoym99) [reply src/icalendar/cal.py]: Ok, @angatha I shall split it into multiple lines for readability.

Comment (pronoym99) [reply src/icalendar/cal.py]: I have done the requested changes. Please have a look @angatha.

**Comment by angatha** [comment]
Thank you for your contribution and cleaning up the codebase.

Can you please also replace the `str.format(...)` call in `src/icalendar/tools.py:33` (and the commented out one in `src/icalendar/parser.py:264`)? Then all these calls are replaced and we do not need a second pull request to get them all.

**Review by angatha — APPROVED** [review]

**Comment by pronoym99** [comment]
@angatha, I have done the requested changes. Please have a look.

**Comment by niccokunzmann** [comment]
Thanks @pronoym99 and @angatha !

**Pull request merged**